### PR TITLE
chore: Migrate `fastify` tests to `node:test`

### LINKED ISSUE #2629: Migrate fastify versioned tests to `node:test`

### PR DESCRIPTION
**Pull request by jsumners-nr**
This PR resolves #2629.

### CODE REVIEW

File: test/versioned/fastify/add-hook.test.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    }
+    assertSegments(transaction.trace.root, expectedSegments)
+
+    txPassed = true
+  })
+
+  await fastify.listen({ port: 0 })
+  const address = fastify.server.address()
+  const result = await common.makeRequest(address, '/add-hook')
+  match(result, { hello: 'world' })
```
Comment (bizob2828) [inline test/versioned/fastify/add-hook.test.js:119]:
```suggestion
  assert.deepEqual(result, { hello: 'world' })
```

File: test/versioned/fastify/add-hook.test.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    assertSegments(transaction.trace.root, expectedSegments)
+
+    txPassed = true
+  })
+
+  await fastify.listen({ port: 0 })
+  const address = fastify.server.address()
+  const result = await common.makeRequest(address, '/error')
+  assert.ok(ok)
+  match(result, {
```
Comment (bizob2828) [inline test/versioned/fastify/add-hook.test.js:180]:
```suggestion
assert.deepEqual(result, {
```

File: test/versioned/fastify/code-level-metrics-hooks.test.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    })
+
+    txPassed = true
+  })
+
+  await fastify.listen({ port: 0 })
+  const address = fastify.server.address()
+  const result = await common.makeRequest(address, '/add-hook')
+
+  match(result, { hello: 'world' })
```
Comment (bizob2828) [inline test/versioned/fastify/code-level-metrics-hooks.test.js:79]:
```suggestion
  assert.deepEqual(result, { hello: 'world' })
```

File: test/versioned/fastify/naming-common.js [hunk 4]
```diff
@@ -36,33 +43,37 @@ module.exports = function createTests(t, getExpectedSegments) {
...
+
+      txPassed = true
     })
+
     await fastify.listen()
     const address = fastify.server.address()
     const result = await makeRequest(address, '/params/id/parent/edit')
-    t.same(result, { id: 'id', parent: 'parent' })
-    t.end()
+    match(result, { id: 'id', parent: 'parent' })
```
Comment (bizob2828) [inline test/versioned/fastify/naming-common.js:75]:
```suggestion
    assert.deepEqual(result, { id: 'id', parent: 'parent' })
```

File: test/versioned/fastify/package.json [hunk 5]
```diff
@@ -12,34 +17,65 @@
...
+        "fastify-gte3-naming.test.js"
       ]
     },
+
     {
       "engines": {
         "node": ">=18"
       },
       "dependencies": {
-        "fastify": ">=3.0.0",
```
Comment (bizob2828) [inline test/versioned/fastify/package.json:28]: we've lost testing any future major versions now. was this actually needed? also this has increased the test matrix a little bit

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: test/versioned/fastify/package.json (hunk 5 above)
Comment (jsumners-nr) [reply test/versioned/fastify/package.json:28]: The block below (line 62 in the updated file) tests `fastify@5`. It's going to be a while before v6 shows up, but we _could_ make that one `>=5.0.0`. 

But, yes, this is needed in order to make sure the right version of the "middie" module is installed that matches the version of Fastify under test. The middie modules utilize Fastify's version matching to assert that it is being loaded into a supported version of Fastify. Without these changes, we end up with version mismatches.

Comment (bizob2828) [reply test/versioned/fastify/package.json:28]: got it. then I would put `>=5.0.0`. so we don't lose test coverage when a new major version is eventually released

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: test/versioned/fastify/package.json (hunk 5 above)
Comment (jsumners-nr) [reply test/versioned/fastify/package.json:28]: Done. But, to be clear, when v6 is released, all of the plugins will also update their compatible versions. So a change here will very likely still be needed.

Comment (bizob2828) [reply test/versioned/fastify/package.json:28]: That's fine

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**